# 3 ways to clearly measure your investment skill (like a professional)

The batting average measures how often an investment manager beats a specific index, like the S&P 500 or Nasdaq.

This index is used as a benchmark that helps evaluate the manager's performance. A higher batting average means the manager often meets or surpasses the benchmark.

This is a sign of good performance. It means the manager is consistently able to outperform their benchmark.

We can use the batting average to measure your own investment performance against a chosen benchmark. Doing this will help determine if you’re better of just investing in the underlying index.

We can use batting average like the professionals with Python.

In [1]:
# Imports and set up

# First, import the tools we’ll need including NumPy, pandas, and the OpenBB SDK.

import numpy as np
import pandas as pd
from openbb_terminal.sdk import openbb

Forecasting Toolkit is disabled. To use the Forecasting features please install the toolkit following the instructions here: https://docs.openbb.co/sdk/quickstart/installation/

Portfolio Optimization Toolkit is disabled. To use the Optimization features please install the toolkit following the instructions here: https://docs.openbb.co/sdk/quickstart/installation/

In [2]:
# Now, download price data and compute returns.

prices = openbb.economy.index([
    "META", 
    "AAPL", 
    "AMZN", 
    "NFLX", 
    "GOOG", 
    "QQQ"
])

returns = prices.pct_change().dropna()
bench_returns = returns.pop("QQQ")
port_returns = (returns * 0.2).sum(axis=1)

In [5]:
returns.head()

META      AAPL      AMZN      NFLX      GOOG
Date                                                        
2012-05-21 -0.109861  0.058260  0.019921  0.025443  0.022835
2012-05-22 -0.089039 -0.007679 -0.012746 -0.056175 -0.021674
2012-05-23  0.032258  0.024400  0.009056  0.062029  0.014414
2012-05-24  0.032187 -0.009184 -0.009389 -0.022806 -0.009517
2012-05-25 -0.033909 -0.005360 -0.010918 -0.000711 -0.020094

In [3]:
# Compute the batting average

# Create a function that computes the batting average. We’ll walk through the code next.

def batting_average(port_returns, bench_returns):
    results = dict(
        {
            "batting average": np.nan,
            "up market": np.nan,
            "down market": np.nan,
        }
    )
    active_returns = port_returns - bench_returns
    ba = active_returns > 0
    up = active_returns[bench_returns >= 0.0] > 0
    down = active_returns[bench_returns < 0.0] > 0

    if len(ba) > 0:
        results["batting average"] = ba.mean()
    if len(up) > 0:
        results["up market"] = up.mean()
    if len(down) > 0:
        results["down market"] = down.mean()

    return pd.Series(results, index=results.keys())

In [4]:
batting_average(port_returns, bench_returns)

batting average    0.506168
up market          0.560807
down market        0.436800
dtype: float64

# Interpret the results

In this example, the batting average is 50.5%. When the benchmark is up, the batting average raises to 56% and when it’s down it falls to 43.7%.

This means the portfolio just meets the minimum threshold of a successful strategy.

Investment managers are paid to outperform the benchmark. The batting average is a great way to determine if their strategy is worth the expense.

For individual investors, active portfolio management comes with risk and opportunity cost. If you were trading this portfolio, you might consider investing in the benchmark instead.